### PR TITLE
Lootable skeleton corpses

### PR DESCRIPTION
**Pull request by Pink-Chink**
## About The Pull Request

Adds equipable and lootable skeleton mob corpses at the request of Cringelord. Updates the ancient corpse spawner with hobo, Excelsior, and One Star presets.

## Why It's Good For The Game

More flavorful method of placing loot for mappers.

## Testing

- Spawned several skeletal corpses with and without loot.
- Extracted bones from skeletons via surgery.
- Equipped and unequipped skeletons.
- Rotated skeletons in all directions.

image

## Credits

Skeleton sprite modified for mob use by Cringelord with edits by SirRichardFrancis/KIROV

## Changelog
:cl:
add: Added skeleton species
add: Added hobo, Excelsior, and One Star corpse spawner presets
tweak: Updated corpse spawner features to allow for more varied corpses
tweak: Added checks for null blood color in blood overlay and blood splatter procs
/:cl:

### CODE REVIEW

**Comment by glaciusinfinity** [comment]
Splendid

**Comment by Teleported-Bread** [comment]
Amazing pr, actually interesting way to distribute loot. 

https://user-images.githubusercontent.com/28495609/205467066-34576929-1b13-4ab8-a881-a5c3e00c8d63.mov

File: code/modules/mob/living/carbon/human/species/station/skeleton.dm [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/datum/species/skeleton
+	name = SPECIES_SKELETON
+	name_plural = "skeletal remains"
+	blurb = "A human of the skeletal variety."	// Could use a neat OS lore bit here
```
Comment (ValoTheValo) [inline code/modules/mob/living/carbon/human/species/station/skeleton.dm:4]: re use the remains description for now
"They look like human remains. Some poor soul expired here, a million miles from home."

Comment (Pink-Chink) [reply code/modules/mob/living/carbon/human/species/station/skeleton.dm:4]: Null Sector is our solar system, right? If so, we should adjust the last part.

Comment (Pink-Chink) [reply code/modules/mob/living/carbon/human/species/station/skeleton.dm:4]: I guess that also depends on the people's perspective of Earth. As far as I know, nobody alive has even seen it. I'll leave the "million miles from home" bit.

File: code/modules/mob/living/carbon/human/human_species.dm [hunk 2]
```diff
@@ -16,3 +16,11 @@
...
 /mob/living/carbon/human/dummy/mannequin/fully_replace_character_name(var/oldname, var/newname)
 	..(newname = "[newname] (mannequin)")
+
+/mob/living/carbon/human/skeleton
+	icon_state = "skeleton"
+
+/mob/living/carbon/human/skeleton/New(new_loc)
+	..(new_loc, SPECIES_SKELETON)
+	STOP_PROCESSING(SSmobs, src)
+	death(FALSE)
```
Comment (SirRichardFrancis) [inline code/modules/mob/living/carbon/human/human_species.dm:26]:
```suggestion
	death(FALSE)
	GLOB.human_mob_list -= src
```

File: code/modules/mob/living/carbon/human/species/station/skeleton.dm [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	blurb = "They look like human remains. Some poor soul expired here, a million miles from home."
+
+	icobase = 'icons/mob/human_races/r_skeleton.dmi'
+	deform = 'icons/mob/human_races/r_skeleton.dmi'
+	damage_overlays = null
+	damage_mask = null
+	blood_mask = null
+
+	min_age = 499	// Oldest skeleton is of a person of 140 years. Implies OS managed to extend life expectancy via technology. Revise according to lore.
+	max_age = 359	// OS disappeared in 2291, CEV Eris launched 2642. This means the skeleton of a child of 8 years would be 359 years old.
```
Comment (SirRichardFrancis) [inline code/modules/mob/living/carbon/human/species/station/skeleton.dm:13]: Are skeletons One Star exclusive?

Comment (glaciusinfinity) [reply code/modules/mob/living/carbon/human/species/station/skeleton.dm:13]: The things I need these for aren't related to onestar, they are still going to see a lot of use in onestar areas but its nice to have some flexibility

Comment (Pink-Chink) [reply code/modules/mob/living/carbon/human/species/station/skeleton.dm]: Alrighty. Skeletons are theme-agnostic and age can be set with the spawner.

File: code/modules/organs/organ_description.dm [hunk 4]
```diff
@@ -196,3 +196,48 @@
...
+	name = "pelvis"
+	max_damage = 20
+	min_broken_damage = 5
+	max_volume = 1
+	default_type = /obj/item/organ/external/skeletal/groin
+
+/datum/organ_description/head/skeletal
+	max_damage = 20
+	min_broken_damage = 5
+	max_volume = 1
```
Comment (TheShown911) [inline code/modules/organs/organ_description.dm:218]: Increase max volume so I can do a skeletal head and brain transplant and become sans undertale.

**Review by TheShown911 — APPROVED** [review]

**Pull request merged**